src/libpmemfile-core/dir.c

```diff
@@ -102,6 +102,17 @@ component_length(const char *path)
 	return (uintptr_t)slash - (uintptr_t)path;
 }
 
+#ifdef DEBUG
+static inline char *
+util_strndup(const char *c, size_t len)
+{
+	char *cp = malloc(len);
+	memcpy(cp, c, len - 1);
```

Comment:
Well, in the only place it's needed, the string we want to copy is exactly len bytes long, so using strncpy will do useless NUL-termination checks for each byte.

FTR, this implementation has a off-by-one bug. I'll push a new version in few minutes.